src/activities/mirrorgame/Mirrorgame.qml

```diff
@@ -0,0 +1,501 @@
+            anchors.horizontalCenter: parent.horizontalCenter
+            PropertyAnimation {
+                id: showOk
+                target: ok
+                properties: "opacity"
+                from: 0
+                to: 1
+                duration: 600
+                }
+            MouseArea {
+                anchors.fill: parent
+                onClicked: {
+                    Activity.win();
+                    lvl1.visible = false;
+                    lvl2.source = "Level2.qml";
+
+
+
+                }
+            }
+        }
+
+        Canvas{
+        id:wire
+        width:parent.width
+        height:parent.height
+        onPaint:{
+            var context = getContext("2d")
+             context.beginPath();
+             context.lineWidth = 4;
+             context.strokeStyle = "black"
+             context.moveTo(torch.x+400,torch.y+10);
+             context.lineTo(torch.x+80,torch.y+10);
+             context.stroke();
+
+        }
+        Image {
+            id: off
+            source:"off.svg"
+            fillMode: Image.PreserveAspectFit
+            x:1180
+            y:140
+            width:100
+            height:100
+            GCText{
+            id:check
+            width:100
+            height:100
+            x:-50
+            y:100
+            text:qsTr("Click to Check");
+            SequentialAnimation on y {
+                loops: Animation.Infinite
+                NumberAnimation {
+                    from:105
+                    to:110
+                    easing.type: Easing.InSine; duration: 1000
+                }
+                NumberAnimation {
+                    from:110
+                    to:105
+                    easing.type: Easing.OutSine; duration: 1000
+                }
+
+            }
+            }
+            MouseArea {
+                anchors.fill: parent
+                onClicked: {check.visible=false;
+                            lvl1.checkAnswer()}
+            }
+        }
+        Image {
+            id: on
+            visible: false
+            source:"on.svg"
+            fillMode: Image.PreserveAspectFit
+            x:1180
+            y:140
+            width:100
+            height:100
+
+        }
+        }
+      function checkAnswer() {
+
+          off.visible=false;
+          on.visible=true;
+           if( mirror1.placed && mirror2.placed && mirror3.placed && mirror4.placed )
+           {solution.opacity=1;
+            bulboff.visible=false;
+            bulbon.visible=true;
+           showOk.start();}
+           else
+           {    on.visible=false;
+               off.visible=true;
+           if(!mirror1.placed)wrongAnswer1.visible=true;
+           if(!mirror2.placed)wrongAnswer2.visible=true;
+           if(!mirror3.placed)wrongAnswer3.visible=true;
+           if(!mirror4.placed)wrongAnswer4.visible=true;
+           }
+            return
+        }
+
+        Image {

```

Comment:
You should probably use a Repeater instead of 4 images with more or less the same code